controllers/rabbitmq/rabbitmq_controller.go

```diff
@@ -471,6 +501,79 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (result ct
 	return ctrl.Result{}, nil
 }
 
+func (r *Reconciler) reconcilePerPodServices(ctx context.Context, instance *rabbitmqv1beta1.RabbitMq, helper *helper.Helper) (ctrl.Result, error) {
+	Log := r.GetLogger(ctx)
+
+	if instance.Spec.PodOverride == nil || len(instance.Spec.PodOverride.Services) == 0 {
+		Log.Info("PodOverride not configured, skipping per-pod service creation")
+		instance.Status.ServiceHostnames = nil
+		return ctrl.Result{}, nil
+	}
+
+	replicas := int(*instance.Spec.Replicas)
+
+	if len(instance.Spec.PodOverride.Services) != replicas {
+		return ctrl.Result{}, fmt.Errorf("number of services in podOverride (%d) must match number of replicas (%d)", len(instance.Spec.PodOverride.Services), replicas)
+	}
+
+	Log.Info("Creating per-pod services using podOverride configuration")
+
+	var serviceHostnames []string
+	for i := 0; i < replicas; i++ {
+		podName := fmt.Sprintf("%s-server-%d", instance.Name, i)
+		svcName := podName
+		serviceOverride := instance.Spec.PodOverride.Services[i]
+
+		svc, err := service.NewService(
+			service.GenericService(&service.GenericServiceDetails{
+				Name:      svcName,
+				Namespace: instance.Namespace,
+				Labels: map[string]string{
+					labels.K8sAppName:      instance.Name,
+					labels.K8sAppComponent: "rabbitmq",
+				},
+				Selector: map[string]string{
+					"statefulset.kubernetes.io/pod-name": podName,
+				},
+				Ports: []corev1.ServicePort{
+					{Name: "amqp", Port: 5672, TargetPort: intstr.FromInt(5672)},
+					{Name: "amqps", Port: 5671, TargetPort: intstr.FromInt(5671)},
+				},
+			}),
+			5,
+			&serviceOverride,
+		)
+		if err != nil {
+			return ctrl.Result{}, fmt.Errorf("error creating service %s: %w", svcName, err)
+		}
+
+		svc.AddAnnotation(map[string]string{
+			networkv1.AnnotationHostnameKey: fmt.Sprintf("%s.%s.svc", svcName, instance.Namespace),
+		})
+
+		ctrlResult, err := svc.CreateOrPatch(ctx, helper)
+		if err != nil {
+			// Check if this is a LoadBalancer IP pending error
+			if errors.Is(err, util.ErrResourceIsNotReady) {
+				// LoadBalancer IP is pending, but service was created successfully
+				// Continue creating remaining services
+				Log.Info(fmt.Sprintf("Service %s LoadBalancer IP pending, continuing with remaining services", svcName))
+			} else {
+				// Real error, return it
+				return ctrlResult, err
+			}
+		} else if (ctrlResult != ctrl.Result{}) {
+			// Service needs requeue but no error
+			Log.Info(fmt.Sprintf("Service %s not ready yet, will requeue", svcName))
+		}
+
+		serviceHostnames = append(serviceHostnames, fmt.Sprintf("%s.%s.svc", svcName, instance.Namespace))
+	}
+
+	instance.Status.ServiceHostnames = serviceHostnames
```

Comment:
if we assign it after the loop, we only get it set if all svc got created successful. wondering if we should update it on each run of the loop so that we e.g. also an update list if 2 out of 3 got successfully reconciled?